lease/lessor.go

```diff
@@ -42,6 +44,13 @@ var (
 	// maximum number of leases to revoke per second; configurable for tests
 	leaseRevokeRate = 1000
 
+	// maximum number of lease checkpoints per second; configurable for tests
+	leaseCheckpointRate = 1000
+
+	// Limit the total number of scheduled checkpoints, checkpoint should be best effort and it is
+	// better to throttle checkpointing than to degrade performance.
+	maxScheduledCheckpoints = 10000
```

Comment:
how do we come up with these default values? have you done any benchmark?

would it be helpful if we make the checkpoint api accept multiple leases as a batch?

Reply:
> how do we come up with these default values? have you done any benchmark?

Not yet, but I need to. I'm betting these number can be much higher. I'll do some benchmarking this week.  For the scheduling, we just need to keep the heap size to some reasonable size, so I might look at typical etcd memory footprints and use that to help establish a limit that is based on the worst case memory utilization we're able to accept.

> would it be helpful if we make the checkpoint api accept multiple leases as a batch?

We just added the batching of lease checkpointing yesterday ([proto change](https://github.com/coreos/etcd/pull/9924/commits/75ac18cd2d94be652c172aac3dbd5485a7533554#diff-c60cc9a0d9c8fe4cdedc8b5944475568R788)) per @gyuho's suggestion.  Since this is not clear from how the `leaseCheckpointRate` constant is defined, so I'll clear that up.  Maybe by defining a `maxLeaseCheckpointBatchSize` and using `leaseCheckpointRate` to define how many patched checkpoint operations can occur per second, which I might set quite low once we have batching.